### ISSUE TITLE
`deprecated` decorator doesn't work when calling deprecated overload as a decorator

### ISSUE THREAD

**Issue by DetachHead** [microsoft/pyright#5948]
**Describe the bug**
when decorating a function with a deprecated overloaded function, the deprecation warning does not appear

**Code or Screenshots**
```python
from typing import TypeVar, overload

from typing_extensions import deprecated

T = TypeVar("T")


@deprecated("asdf")
@overload
def foo(value: T) -> T: ...


@overload
def foo(value: str): ...


def foo(value: object) -> object: ...


foo(1)  # deprecation warning
foo("")  # deprecation warning


@foo  # no deprecation warning
def asdf(): ...
```
![image](https://github.com/microsoft/pyright/assets/57028336/42ef143c-6cf4-4f6d-9d78-5296526dd903)

**VS Code extension or command-line**
pylance 2023.9.10

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.328, which I just published. It will also be included in a future release of pylance.
